sql/catalyst/src/main/scala/org/apache/spark/sql/types/UserDefinedType.scala

```diff
@@ -50,13 +54,46 @@ abstract class UserDefinedType[UserType >: Null] extends DataType with Serializa
   /** Serialized Python UDT class, if exists. */
   def serializedPyClass: String = null
 
-  /**
-   * Convert the user type to a SQL datum
-   */
-  def serialize(obj: UserType): Any
+  /** The RowEncoder used to serialize/deserialize extenal row to internal row format. */
+  private lazy val rowEncoder = UserDefinedType.getRowEncoder(sqlType)
```

Comment:
Due to recently added constraint that SQLConf can't be accessed during task execution, this causes the test failure. I'm wondering if we have other options here...